patch.yml

```diff
@@ -13,6 +13,54 @@
 # limitations under the License.
 
 ---
+- name: Play | Set correct RDBMS release and set fact if this is a single node inventory
+  hosts: dbasm
+  gather_facts: false
+  tasks:
+    - name: include defaults
+      include_vars:
+        file: roles/common/defaults/main.yml
+
+    - name: printing oracle_rel
+      debug:
+        msg:
+          - "oracle_rel b4 is : {{ oracle_rel }}"
+
+    - name: printing gi_patches piped into json_query
+      debug:
+        msg:
+          - "Loop variables are: {{ gi_patches | json_query('[?base==`' + oracle_ver + '`].release') }}"
+
+    - name: Get the highest version number from gi_patches array
+      set_fact:
+        highest_version_num: "{% if not ansible_loop.first %}{% if item is version(highest_version_num, '>=') %}{{ item }}{% else %}{{ highest_version_num }}{% endif %}{% else %}{{ item }}{% endif %}"
+      loop: "{{ gi_patches | json_query('[?base==`' + oracle_ver + '`].release') }}"
+      loop_control:
+        extended: yes
+      when:
+        - oracle_rel is defined
+        - oracle_rel != "base"
+
+    - name: printing highest_version_num
+      debug:
+        msg:
+          - "highest_version_num is : {{ highest_version_num }}"
+
+    - name: Set oracle_rel to highest_version_num if oracle_rel == 'latest'
+      set_fact:
+        oracle_rel: "{{ highest_version_num }}"
+      when:
+        - oracle_rel == 'latest'
+
+    - name: set fact is this is a single node inventory - needed for RAC patching (MOS note 2336186.1)
```

Comment:
The output here can be misleading:  even with multi-node, it reports as OK rather than skipped.  What do you think about adding a conditional in the patch command line directly, just like we do for ocm:

{{ '-ocmrf {{ swlib_unzip_path }}/ocm.rsp' if patch.ocm else '' }}

It might allow us to cut down this ~45 new lines of code by a fair bit.

Reply:
We don't want it to be skipped and to avoid the misleading interpretation, I think the next segment of print block is very helpful.
```
    - name: set fact is this is a single node inventory - needed for RAC patching (MOS note 2336186.1)
      set_fact:
        non_rolling_clause: "{{ ' -nonrolling' if groups['dbasm'] | length==1 else ' '}}"
```